HW05.Rmd

```diff
@@ -0,0 +1,158 @@
+---
+title: "HW05"
+author: "Mingyu Qi"
+date: "8/3/2020"
+output: github_document
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+# 1. Tidying the dad_mom file
+
+## Read in dad_mom file
+
+```{r read_1, echo=TRUE, message=FALSE, warning=FALSE}
+#load libraries
+library(tidyverse)    
+library(knitr)
+library(ggrepel)
+
+#read in "dad_mom.txt" file
+dad_mom <- read.delim("dad_mom.txt")
+glimpse(dad_mom)
+```
+
+
+## Clean up dad_mom data frame 
+
+```{r clean, echo=TRUE, message=FALSE, warning=FALSE}
+dad_mom_tidy <- dad_mom %>%
+  #use gather() to create a "dad or mom" indicator and a name variable 
+  gather(key = "parent", value = "name", name_dad,name_mom) %>%
+  #re-code variable dad_mom to parent which equals "dad" or "mom"
+  mutate(parent = recode(parent, 'name_dad' = "dad", .default = "mom")) %>%
+  #use ifelse() to create a new income variable 
+  mutate(income = ifelse(parent == "dad", income_dad, income_mom)) %>%
+  #drop old variables 
+  select(-income_dad, -income_mom) %>%
+  #sort data set by family ID
+  arrange(fam_id)
+
+#print data set 
+print(dad_mom_tidy)
+```
+
+
+
+# 2. Joining together CTRP data
+
+## Read in five CSV files
+
+```{r read_2, echo=TRUE, message=FALSE, warning=FALSE}
+#read in AUC_1
+AUC_1 <- read.csv("CTRP_files/AUC_1.csv")
+
+#read in AUC_2
+AUC_2 <- read.csv("CTRP_files/AUC_2.csv")
+
+#read in cancer_cell_line_info
+cancer_cell_line_info <- read.csv("CTRP_files/cancer_cell_line_info.csv")
+
+#read in compound_info
+compound_info <- read.csv("CTRP_files/compound_info.csv")
+
+#read in Experiment_info
+Experiment_info <- read.csv("CTRP_files/Experiment_info.csv")
+```
+
+
+## Join data sets together
+
+```{r read, echo=TRUE, message=FALSE, warning=FALSE}
+#Vertically join AUC_1 and AUC_2
+AUC <- bind_rows(AUC_1, AUC_2) %>%
+  arrange(experiment_id)
+
+#Join with Experiment_info
+AUC_Exp <- AUC %>%
+  full_join(Experiment_info, by = c("experiment_id" = "expt_id"))
+
+#Join with cancer_cell_line_info
+AUC_Cancer <- AUC_Exp %>%
+  full_join(cancer_cell_line_info, by = "master_ccl_id")
+
+#Join with compound_info
+AUC_Compound <- AUC_Cancer %>%
+  full_join(compound_info, by = "master_cpd_id")
+```
+
+
+## Which cancer type has the lowest AUC values to the compound "vorinostat"?
+
+```{r graph_1, echo=TRUE, message=FALSE, warning=FALSE}
+#limit data to the compound "vorinostat" and summarize average AUC of each cancer type
+AUC_vor <- AUC_Compound %>%
+  filter(cpd_name == "vorinostat" & is.na(cancer_type) == FALSE & is.na(area_under_curve) == FALSE ) %>%
+  group_by(cancer_type) %>%
```

Comment:
I kept the data where cancer type is missing in my plot, and I did not  check for missing AUC values. I wonder how much difference these make.